### ISSUE TITLE
flutter pub prints debug-like output

### ISSUE THREAD

**Issue by mit-mit** [flutter/flutter#118314]
When `flutter pub` fails, it prints the expected pub failure in white, but then after that some internal-looking, debug-like output in screaming red. Can we remove then red output (or at least only print it when running with `--verbose`)?

![unnamed](https://user-images.githubusercontent.com/13644170/211802473-b406ebe9-7344-4e2f-9a7d-5da4191f4ae0.png)

**Comment by exaby73** [comment]
Hello @mit-mit. Can you provide some reproduction steps for this, along with a [complete and minimal, reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) if applicable?

**Comment by mit-mit** [comment]
It happens on a number of pub issues.

Here's one example (with latest flutter main):
1. `flutter create testapp`
2. `cd testapp`
3. Edit `pubspec.yaml` and change the SDK constraint to `sdk: '>=2.0.0 <3.0.0'`
4. `flutter pub get`

**Comment by exaby73** [comment]
## Triage report

Using the above mentioned steps, I can reproduce this issue on `Master (3.7.0-19.0.pre.13)`

```console
Resolving dependencies...
The current Dart SDK version is 3.0.0-112.0.dev.

Because issue_118314 doesn't support null safety, version solving failed.

The lower bound of "sdk: '>=2.0.0 <3.0.0'" must be 2.12.0 or higher to enable null safety.
For details, see https://dart.dev/null-safety
pub failed
command: "/Users/nabeelparkar/fvm/versions/master/bin/cache/dart-sdk/bin/dart __deprecated_pub get --example --directory ."
pub env: {
  "FLUTTER_ROOT": "/Users/nabeelparkar/fvm/versions/master",
  "PUB_ENVIRONMENT": "flutter_cli:get",
  "PUB_CACHE": "/Users/nabeelparkar/.pub-cache",
}
exit code: 1
```

<details><summary><code>flutter doctor -v</code> (Master)</summary>

```
[!] Flutter (Channel master, 3.7.0-19.0.pre.13, on macOS 13.0.1 22A400 darwin-arm64, locale en-US)
    • Flutter version 3.7.0-19.0.pre.13 on channel master at /Users/nabeelparkar/fvm/versions/master
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/2.18.6/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/nabeelparkar/fvm/versions/master. Consider adding /Users/nabeelparkar/fvm/versions/master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f8628b5cb1 (4 hours ago), 2023-01-11 18:48:03 -0800
    • Engine revision 4a8d6866a1
    • Dart version 3.0.0 (build 3.0.0-112.0.dev)
    • DevTools version 2.20.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/nabeelparkar/Library/Android/sdk/
    • Platform android-33, build-tools 33.0.0
    • ANDROID_SDK_ROOT = /Users/nabeelparkar/Library/Android/sdk/
    • Java binary at: /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /Applications/Brave Browser.app/Contents/MacOS/Brave Browser

[✓] Android Studio (version 2021.3)
    • Android Studio at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.1)
    • IntelliJ at /Users/nabeelparkar/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 71.3.6
    • Dart plugin version 223.8214.16

[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.1)
    • IntelliJ at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/223.8214.52/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.56.0

[✓] Connected device (4 available)
    • iPhone 14 Pro (mobile)     • A75A00DB-CD34-4A0D-BFD9-2DA753E5A5EE • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-2 (simulator)
    • iPhone 14 Pro Max (mobile) • 6BC61E88-4DE0-4BCB-8599-DA5294269146 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-2 (simulator)
    • macOS (desktop)            • macos                                • darwin-arm64   • macOS 13.0.1 22A400 darwin-arm64
    • Chrome (web)               • chrome                               • web-javascript • Brave Browser 108.1.46.153

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>
